core/src/main/java/org/elasticsearch/common/settings/Settings.java

```diff
@@ -766,7 +770,15 @@ public String get(String key) {
                 return retVal;
             }
             // try camel case version
-            return map.get(toCamelCase(key));
+            String camelKey = toCamelCase(key);
+            if (key.equals(camelKey)) {
+                return null;
+            }
```

Comment:
we don't need to compare keys here and could just rely on the lookup returning a non-null value?